[BUG] CI: typo goto-bus/setup-zig → 404, blocks every PR

# 🔧 [BUG] CI: typo `goto-bus/setup-zig` → 404, blocks every PR (incl. BUG-001 #27)

**Classification:** P0-CRITICAL (blocks zgf#27 BUG-001-BF16 fix → blocks IGLA Number Race)
**Kingdom:** Network (CI infra)
**Soul-name suggested:** `ZigPathFixer`

---

## 🎯 Goal

Replace the dead `goto-bus/setup-zig@v2` reference in [`.github/workflows/test-bindings.yml`](https://github.com/gHashTag/zig-golden-float/blob/main/.github/workflows/test-bindings.yml) with the actively-maintained canonical action.

## 🐛 Symptom (verified via mcp.github)

PR #27 (and every prior PR in zgf using this workflow) fails 5 jobs identically:

```
##[error]Unable to resolve action goto-bus/setup-zig, repository not found
```

Affected runs (sample):
- [run 25112441466](https://github.com/gHashTag/zig-golden-float/actions/runs/25112441466) on PR #27
- jobs 73589940553, 73589940602, 73589940607, 73589940626, 73589940644 — all the same root cause

## 🔍 Root cause (R5-honest, line-pinned)

Two workflow files reference the upstream action — one correct, one typo:

| File | Reference | Exists? |
|------|-----------|---------|
| `.github/workflows/release.yml` | `goto-bus-stop/setup-zig@v2` | ✅ exists (157 stars, last push 2024-09-28, NOT archived) |
| `.github/workflows/test-bindings.yml` | `goto-bus/setup-zig@v2` | ❌ **404 — typo, missing `-stop`** |

The "goto-bus" account has nothing — it's a phantom reference. This is a single-character (well, suffix) typo from `goto-bus-stop`.

## 📋 Acceptance Criteria

```
- [ ] Patch .github/workflows/test-bindings.yml line(s) using setup-zig:
       Either (a) fix typo to existing fork: `goto-bus-stop/setup-zig@v2`, or
       (b) migrate to actively-maintained: `mlugg/setup-zig@v2`.

- [ ] If (b), also patch .github/workflows/release.yml for consistency
       — keep all zgf workflows on the same Zig setup action.

- [ ] On a fresh PR or push to a feature branch, "Test Multi-Language Bindings"
       workflow proceeds past "Set up job" / "Install Zig" without a 404 error.

- [ ] PR #27 (BUG-001-BF16 fix) re-runs CI green AFTER this PR merges.

- [ ] L7 ledger row via mcp.tri_railway.railway_experience_append phi_step=COMMIT.
```

## 🚫 Non-Goals

- Do NOT change the Zig compiler version pinned by the workflow.
- Do NOT change anything else in the workflow other than the action reference.
- Do NOT introduce a third Zig setup action — converge on ONE.

## 💡 Recommendation

**Path B — migrate to `mlugg/setup-zig@v2`.**

| Action | Pros | Cons |
|---|---|---|
| `goto-bus-stop/setup-zig@v2` | minimal diff (just fix typo) | last push 2024-09-28, no Zig 0.16 testing |
| **`mlugg/setup-zig@v2`** | **active (231 stars, last push 2026-01-19), explicit Zig 0.16 support** | small migration touch in 2 files |
| `korandoru/setup-zig` | none over alternatives | archived 2024-10-05 |

`mlugg/setup-zig` has identical input schema (`with: { version: ... }`) so the migration is literally:

```diff
- uses: goto-bus/setup-zig@v2
+ uses: mlugg/setup-zig@v2
```

(plus the same change in `release.yml`).

## ⏱️ Time budget

- Patch + commit: ~2 min
- Push + open PR: ~30 sec
- CI validation on the same PR: ~5 min
- **Total: ~10 min from claim to merged**

## 🔗 Cross-links

- Blocks [zgf#27](https://github.com/gHashTag/zig-golden-float/pull/27) (BUG-001-BF16 fix patch verified canonical, only blocked on this CI infra)
- Blocks [zgf#26 v2 BENCH-007b re-run](https://github.com/gHashTag/zig-golden-float/issues/26) bf16 column splice
- Blocks [zgf#25 race execution order](https://github.com/gHashTag/zig-golden-float/issues/25) Step 0
- Blocks [trios#143](https://github.com/gHashTag/trios/issues/143) leaderboard publication
- No conflict with Phase 1 promotion path (different repo)

## 📜 Anchor

`phi^2 + phi^-2 = 3 · TRINITY · NEVER STOP`

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -40,7 +40,7 @@ jobs:
         uses: actions/checkout@v4
 
       - name: Setup Zig
-        uses: goto-bus-stop/setup-zig@v2
+        uses: mlugg/setup-zig@v2
         with:
           version: 0.15.0
 

--- a/.github/workflows/test-bindings.yml
+++ b/.github/workflows/test-bindings.yml
@@ -31,7 +31,7 @@ jobs:
         uses: actions/checkout@v4
 
       - name: Install Zig
-        uses: goto-bus/setup-zig@v2
+        uses: mlugg/setup-zig@v2
         with:
           version: 0.15.2
 

